### PR TITLE
[RISCV][VLOPT] Add vector mask producing integer instructions to isSupportedInstr and getOperandInfo

### PR DESCRIPTION
**Pull request by michaelmaitland**

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp [hunk 1]
```diff
@@ -438,7 +444,11 @@ static OperandInfo getOperandInfo(const MachineInstr &MI,
   case RISCV::VWMACC_VX:
   case RISCV::VWMACCSU_VV:
   case RISCV::VWMACCSU_VX:
-  case RISCV::VWMACCUS_VX: {
+  case RISCV::VWMACCUS_VX:
+  // Vector Single-Width Fractional Multiply with Rounding and Saturation
+  // Destination EEW=2*SEW and EMUL=2*EMUL. Source EEW=SEW and EMUL=LMUL.
+  case RISCV::VSMUL_VV:
```
Comment (topperc) [inline llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp:450]: vsmul doesn't produce a 2*sew result. It does internally and then shift and truncates it to sew

Comment (michaelmaitland) [reply llvm/lib/Target/RISCV/RISCVVLOptimizer.cpp:450]: Sorry, this wasn't supposed to be a part of this patch anyway. Thanks for the info though.

File: llvm/test/CodeGen/RISCV/rvv/vl-opt-op-info.mir [hunk 2]
```diff
@@ -602,4 +602,83 @@ body: |
...
 ...
-
+---
+name: vmop_vv
+body: |
+  bb.0:
+    ; CHECK-LABEL: name: vmop_vv
+    ; CHECK: %x:vr = PseudoVMSEQ_VV_M1 $noreg, $noreg, 1, 0 /* e8 */
+    ; CHECK-NEXT: %y:vr = PseudoVMAND_MM_M1 $noreg, %x, 1, 0 /* e8 */
+    %x:vr = PseudoVMSEQ_VV_M1 $noreg, $noreg, -1, 0
```
Comment (topperc) [inline llvm/test/CodeGen/RISCV/rvv/vl-opt-op-info.mir:612]: The EEW for a VMSEQ must be 3, 4, 5, or 6.

Comment (michaelmaitland) [reply llvm/test/CodeGen/RISCV/rvv/vl-opt-op-info.mir:612]: Thanks. Updated. Is that something we could enforce in the MachineVerifier?

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**